### DISCUSSION TITLE
🚀 The Future of Potter DB

### DISCUSSION THREAD

**Discussion by danielschuster-muc** [danielschuster-muc/potter-db#338] [Announcements]
## 📜 Background 
Potter DB started in mid July 2022 as a fun project of mine. As a big Harry Potter Fan I thought an API with a big data collection from the Harry Potter Universe was missing. 

But before opening the API to public and releasing it, much work had to be done:

- collecting the data from sources like the Harry Potter Wiki, the books, movies and other sources
- starting on the api and working on a system to effectively import the data into it
- creating the Potter DB Website to view and search data

End of September I opened Potter DB to public: Meaning everyone could use the API and contribute to it.
During Hacktoberfest 2022 more than 10 people contributed to the project, we suddenly had plenty of new pages for the DB Search, more data and much more. Definitely a huge success!

### Chapter Summaries
The [chapter summaries](https://api.potterdb.com/v1/books/harry-potter-and-the-philosopher-s-stone/chapters) of the books are still empty and therefore I am **looking for helpers** to write short summaries for each chapter.  @kat-kan is currently working on the chapters for the first book. As a reference for content and length you can checkout [this discussion](https://github.com/danielschuster-muc/potter-db/discussions/170). 

### GraphQL API
Our GraphQL API is still missing some important features such as filtering and sorting. If you have any ideas on how to implement and improve it, write those down [here](https://github.com/danielschuster-muc/potter-db/discussions/226). 

### Documentation 
We also have a new [documentation](https://docs.potterdb.com/) for the Potter DB API. But it needs improvements in detail and language and should also be multilingual. 

### Quotes
A new planned feature for the API is an endpoint to list and view quotes from the Potter Universe. I will probably make a basic setup for it and afterwards people can a) add more quotes and b) create the frontend pages for the Website. 

### TypeScript
We should definitely add TypeScript to the frontend part to add static typing and prevent other smaller issues. As this is a big task and will change a lot please wait before implementing it directly. I will probably either do it myself or create a detailed plan for it. 

### Next.js 13 Features
With [Next.js v13](https://nextjs.org/blog/next-13) there are a lot of new features and changes we should make. 
This includes: 
- app directory (new project structure) 
- new next/image 
- turbopack for faster startup
When and how we implement those will be explained later on.

### OG Image Generation
Another feature we should implement in the future is the relatively new [OG Image Generation](https://vercel.com/docs/concepts/functions/edge-functions/og-image-generation) by Vercel. This allows us to easily create dynamic social media images. 

### Open Issue
There are also some [open issue](https://github.com/danielschuster-muc/potter-db/issues?q=is%3Aissue+is%3Aopen+no%3Aassignee) you can immediately start working on. Open a issue if you found a bug or want to suggest an new feature. 

## 🏹 Conclusion
Thank you very much for reading this. I hope it clears some questions about the future of Potter DB. If you have any further ideas or thoughts related to one of those above mentioned topics, feel free to leave a comment below. 

Thanks for helping and big shoutout to all those amazing people who contributed to Potter DB ❤️!

**Comment by joaovitormp1998** [comment]
Pq as apis estão fora no heroku

**Reply by danielschuster-muc** [reply]
You mean why we use heroku for the API? Why not? It works good and it is easy to handle.

**Comment by joaovitormp1998** [comment]
![image](https://user-images.githubusercontent.com/49036365/206021907-8091cfbd-317f-4d21-9c04-fae3ad728167.png)

**Reply by danielschuster-muc** [reply]
That’s not my api. I am not in any relation with that api.

The potterdb api is found at https://api.potterdb.com.
